apps/api/src/main.ts

```diff
@@ -1,14 +1,39 @@
+require('dotenv').config();
+import cors from 'cors';
 import express from 'express';
+import helmet from 'helmet';
+import mongoose from 'mongoose';
+import { corsOptions } from './config/corsConfig';
+import routerCatergories from './routes/categories';
+import postsRouter from './routes/posts';
+import { verifyToken } from './middleware/auth';
+import { errorHandler } from './middleware/errorHandler';
+import auth from './routes/auth';
 
 const host = process.env.HOST ?? 'localhost';
 const port = process.env.PORT ? Number(process.env.PORT) : 3000;
 
 const app = express();
 
-app.get('/', (req, res) => {
-  res.send({ message: 'Hello MFEE!' });
-});
+app.use(express.json());
+app.use(helmet());
+app.use(cors(corsOptions));
 
-app.listen(port, host, () => {
-  console.log(`[ ready ] http://${host}:${port}`);
-});
+app.use('/api/categories', routerCatergories);
+app.use('/api/posts', postsRouter);
+app.use('/api/auth', auth);
+
+app.use(verifyToken);
```

Comment:
If we want to validate the token this need to be before our endpoints, it should look like
```
app.use(verifyToken);

app.use('/api/categories', routerCatergories);
app.use('/api/posts', postsRouter);
app.use('/api/auth', auth);
```